docs/proposals/20230123-azure-service-operator.md

```diff
@@ -0,0 +1,252 @@
+  - "/docs/proposals/20210716-async-azure-resource-creation-deletion.md"
+---
+
+# Managing Azure Resources with Azure Service Operator
+
+## Table of Contents
+
+- [Managing Azure Resources with Azure Service Operator](#managing-azure-resources-with-azure-service-operator)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1 - Users of new Azure features](#story-1---users-of-new-azure-features)
+      - [Story 2 - Developer experience](#story-2---developer-experience)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+        - [Reconciliation](#reconciliation)
+        - [API Changes](#api-changes)
+        - [Installation](#installation)
+        - [Caveats](#caveats)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+      - [Adopt new SDK directly](#adopt-new-sdk-directly)
+      - [Crossplane](#crossplane)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+- **Azure Service Operator (ASO)** - ([link](https://azure.github.io/azure-service-operator/)) a collection of Kubernetes CRDs and controllers which extend the Kubernetes API to manage Azure resources like virtual machines and resource groups.
+- **Bring-your-own (BYO) resource** - an Azure resource backing a Cluster API cluster but managed by the user. CAPI and CAPZ should not modify or delete this resource.
+
+## Summary
+
+This proposal describes how CAPZ could utilize Azure Service Operator (ASO) to create, read, update, and delete Azure resources with the Kubernetes API and custom resources defined and reconciled by ASO.
+
+## Motivation
+
+As a Cluster API infrastructure provider, CAPZ's functionality can be broadly divided into:
+1. Translating Cluster API resource definitions into Azure infrastructure definitions
+2. Interfacing with the Azure platform to manage creating, updating, and deleting that infrastructure
+
+Currently, CAPZ handles `2.` with Azure API calls, which requires ensuring dependencies between Azure resources are met and maintaining state to asynchronously handle long-running operations. Because ASO's primary objective is to handle `2.`, CAPZ could then more declaratively specify all of the infrastructure needed for a given Cluster API resource and let ASO handle the finer details of ensuring that desired state is achieved in Azure. By then primarily focusing on `1.`, CAPZ reduces its overall code maintenance burden and affirms ASO as the canonical way to manage Azure infrastructure from Kubernetes.
```

Comment:
Where we talk about "CAPZ could then more declaratively specify..." let's be clear we're talking about the implementation details of the reconciler controllers (CAPZ already declaratively specifies infrastructure via its APIs, so we don't want to suggest that this proposal will change the APIs).

How about "CAPZ's controller code with the help of Kubernetes's standard client-go library could then communicate with ASO resources declaratively and let the ASO controllers handle the finer details of ensuring that desired state is achieved in Azure."